### PR TITLE
Issue296 comment code gen

### PR DESCRIPTION
**Pull request by shahrzadav**
# Description

This PR provides header and java docs for all code generator templates and displays them after generation.

Fixes # 296

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Test Configuration**:
* Eclipse Version: 2019-06
* Java Version: 1.8
* OS: Windows 10

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by shahrzadav** [comment]
@schlichtig @enriozuni cause of some unsigned commits, it cannot be merged. Please review this branch, and later I will make a new one without any merge problems.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybrid/SecureEncryptor.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /********************************************************************************
- * Copyright (c) 2015-2019 TU Darmstadt, Paderborn University
+ * Copyright (c) 2015-2021 TU Darmstadt, Paderborn University
```
Comment (enriozuni) [inline plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybrid/SecureEncryptor.java:2]: I was trying to generate this encryption hybrid template through CogniCrypt GEN, but did not find the right combination of options to do so. Which options shall I choose to do so and test its generation? Thanks

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybridfiles/SecureEncryptor.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /********************************************************************************
- * Copyright (c) 2015-2019 TU Darmstadt, Paderborn University
+ * Copyright (c) 2015-2021 TU Darmstadt, Paderborn University
```
Comment (enriozuni) [inline plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybridfiles/SecureEncryptor.java:2]: Same thing here as well. I am not finding the right combination of options in the wizard in order to generate this template and test its generation. Thanks if you could help me.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java [hunk 3]
```diff
@@ -16,20 +16,22 @@
 
 import de.cognicrypt.codegenerator.crysl.CrySLCodeGenerator;
 
+
```
Comment (enriozuni) [inline plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java:19]: I suggest adding a simple Javadoc for the class, as you have done for the other templates as well. Also, which task in CogniCrypt GEN, generates this template? I could not find it myself.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java [hunk 4]
```diff
@@ -16,20 +16,22 @@
...
+	
 	public static java.lang.String createHash(java.lang.String msg) throws GeneralSecurityException {
 		byte[] plainBytes = msg.getBytes(StandardCharsets.UTF_8);
 		byte[] out = null;
 		CrySLCodeGenerator.getInstance().includeClass("java.security.MessageDigest").addParameter(out, "out").generate();
 		return Base64.getEncoder().encodeToString(out);
 	}
 
-	public static boolean verifyHash(java.lang.String compareeHash, java.lang.String newMsg) {
+	public static boolean verifyHash(java.lang.String compareHash, java.lang.String newMsg) {
```
Comment (enriozuni) [inline plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java:30]: I would also suggest adding Javadocs in both methods in this class, `createHash()` and `verifyHash()`, similar to how it is done in other templates.

**Review by enriozuni — COMMENTED** [review]
Sorry for the delay in reviewing this. I checked all your comments and Javadocs and also generated each template to check whether the Javadocs are correctly generated. I think that the documentation is very good. Everything is working fine when generated, besides the templates that I could not find the right combination of options in the wizard, to generate the respective templates, so I would appreciate it if you could lead me in that :)

Also, check the other comments regarding the missing Javadocs in the one template.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybrid/SecureEncryptor.java (hunk 1 above)
Comment (shahrzadav) [reply plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybrid/SecureEncryptor.java:2]: Thanks, @enriozuni, for reviewing my PR. In order to reach this, you have to answer the first question "Unencrypted digital channel (e.g. email)" and the second one Byte array or other.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybridfiles/SecureEncryptor.java (hunk 2 above)
Comment (shahrzadav) [reply plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/encryptionhybridfiles/SecureEncryptor.java:2]: In this case, answer the first question, Unencrypted digital channel (e.g., email) and the second one, file.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java (hunk 3 above)
Comment (shahrzadav) [reply plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java:19]: Unfortunately, I could not find a use case in the codegen wizard that leads to the respective template. I talked to Krishna about it, and we decided to leave it for now. That's why I did not add any java docs here.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java (hunk 4 above)
Comment (shahrzadav) [reply plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java:30]: same thing here.

File: plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java (hunk 3 above)
Comment (enriozuni) [reply plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/crysl/templates/stringhashing/StringHasher.java:19]: Oh OK. I also could not find a use case in the wizard where this template is generated. Maybe we can decide to remove it in the future.

**Pull request closed without merging**